clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -24,9 +27,135 @@ CIRGenModule::CIRGenModule(mlir::MLIRContext &context,
                            clang::ASTContext &astctx,
                            const clang::CodeGenOptions &cgo,
                            DiagnosticsEngine &diags)
-    : astCtx(astctx), langOpts(astctx.getLangOpts()),
-      theModule{mlir::ModuleOp::create(mlir::UnknownLoc())},
-      target(astCtx.getTargetInfo()) {}
+    : builder(&context), astCtx(astctx), langOpts(astctx.getLangOpts()),
+      theModule{mlir::ModuleOp::create(mlir::UnknownLoc::get(&context))},
+      diags(diags), target(astCtx.getTargetInfo()) {}
+
+mlir::Location CIRGenModule::getLoc(SourceLocation cLoc) {
+  assert(cLoc.isValid() && "expected valid source location");
+  const SourceManager &sm = astCtx.getSourceManager();
+  PresumedLoc pLoc = sm.getPresumedLoc(cLoc);
+  StringRef filename = pLoc.getFilename();
+  return mlir::FileLineColLoc::get(builder.getStringAttr(filename),
+                                   pLoc.getLine(), pLoc.getColumn());
+}
+
+mlir::Location CIRGenModule::getLoc(SourceRange cRange) {
+  assert(cRange.isValid() && "expected a valid source range");
+  mlir::Location begin = getLoc(cRange.getBegin());
+  mlir::Location end = getLoc(cRange.getEnd());
+  mlir::Attribute metadata;
+  return mlir::FusedLoc::get({begin, end}, metadata, builder.getContext());
+}
+
+void CIRGenModule::buildGlobal(clang::GlobalDecl gd) {
+  const auto *global = cast<ValueDecl>(gd.getDecl());
+
+  if (const auto *fd = dyn_cast<FunctionDecl>(global)) {
+    // Update deferred annotations with the latest declaration if the function
+    // was already used or defined.
+    if (fd->hasAttr<AnnotateAttr>()) {
+      errorNYI(fd->getSourceRange(), "defferedAnnotations");
```

Comment:
Typo: `deferredAnnotations`

Also, the curlies should be removed per coding style (and local convention).

Reply:
> the curlies should be removed per coding style

Ugh.  One constant in my 30+ years of programming is to always use `{ }` for sub-statements.  Code without the curlies just looks wrong and takes more effort to read.  It's hard for me to not fix code that doesn't use `{ }`.  It will be even harder for me to not use `{ }` in code that I write.

Reply:
>Ugh. One constant in my 30+ years of programming is to always use { } for sub-statements. Code without the curlies just looks wrong and takes more effort to read. It's hard for me to not fix code that doesn't use { }. It will be even harder for me to not use { } in code that I write.

This was definitely one of my  harder parts of the coding standard to get used to :)  I managed to relax the rule a bit at least, but this is effectively the canonical example of 'skip curleys'.

Reply:
Fixed the typo and removed the curly braces.